### PR TITLE
add Dockerfile

### PR DESCRIPTION
**Pull request by Yam0r**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,9 @@
+POSTGRES_USER=postgres
+POSTGRES_PASSWORD=123456
+POSTGRES_DATABASE=blog
+POSTGRES_LOCAL_PORT=5434
+POSTGRES_DOCKER_PORT=5432
+
+SPRING_LOCAL_PORT=8088
+SPRING_DOCKER_PORT=8080
+DEBUG_PORT=5005
```
Comment (liliia-ponomarenko) [inline .env]: Don’t push the `.env` file with sensitive data. better create `.env.templete`.
Let’s add `.env` to `.gitignore`

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  postgresdb:
+    platform: linux/arm64
+    image: postgres
+    restart: unless-stopped
+    env_file: ./.env
+    environment:
+      - POSTGRES_USER=$POSTGRES_USER
+      - POSTGRES_PASSWORD=$POSTGRES_PASSWORD
+      - POSTGRES_DB=$POSTGRES_DATABASE
+    ports:
```
Comment (liliia-ponomarenko) [inline docker-compose.yml]: Let’s  use MySQL in your setups

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      - "5433:5423"
```
Comment (liliia-ponomarenko) [inline docker-compose.yml]: use variables

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    environment:
+      SPRING_APPLICATION_JSON: '{
+        "spring.datasource.url"  : "jdbc:postgresql://postgresdb:$POSTGRES_DOCKER_PORT/$POSTGRES_DATABASE",
+        "spring.datasource.username" : "$POSTGRES_USER",
+        "spring.datasource.password" : "$POSTGRES_PASSWORD",
+        "spring.jpa.properties.hibernate.dialect" : "org.hibernate.dialect.PostgreSQLDialect"
+      }'
+      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005"
```
Comment (liliia-ponomarenko) [inline docker-compose.yml]: the same use variable for the debug port value

File: pom.xml [hunk 5]
```diff
@@ -136,6 +136,16 @@
 			<version>2.0.1.Final</version>
 		</dependency>
 
+		<dependency>
+			<groupId>org.postgresql</groupId>
+			<artifactId>postgresql</artifactId>
+		</dependency>
```
Comment (liliia-ponomarenko) [inline pom.xml]: so which DB do you use?

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 6]
```diff
@@ -13,4 +13,19 @@ databaseChangeLog:
...
+#  - include:
+#      file: db/changelog/changes/create-users-roles-table.yaml
+#  - include:
+#      file: db/changelog/changes/insert-roles.yaml
+#  - include:
+#      file: db/changelog/changes/insert-roles-of-users.yaml
+#  - include:
+#      file: db/changelog/changes/insert-users.yaml
+#  - include:
+#      file: db/changelog/changes/set-bcrypt-password.yaml
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: Let’s  uncomment changesets

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  improve your solution ;)

File: .env [hunk 7]
```diff
@@ -0,0 +1,7 @@
+MYSQL_ROOT_PASSWORD=root
```
Comment (okuzan) [inline .env]:
```suggestion
MYSQL_ROOT_PASSWORD=
```

Comment (okuzan) [reply .env]: not fixed, push .env.template here with actual values removed
use git rm --cached .env to remove .env from remote while keeping it locally, and ensure it's not committed again by adding it to .gitignore

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+services:
+  mysqldb:
+    image: mysql:8.0
+    restart: unless-stopped
+    env_file: .env
+    environment:
+      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
+      - MYSQL_DATABASE=${MYSQL_DATABASE}
+    ports:
+      - "3307:3306"
```
Comment (okuzan) [inline docker-compose.yml]: not fixed, use ${ENV_VARIABLES} for ports, same for debug port at the last line

**Review by okuzan — CHANGES_REQUESTED** [review]

File: docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+      - "${SPRING_LOCAL_PORT}:${SPRING_DOCKER_PORT}"
+    environment:
+      SPRING_APPLICATION_JSON: >-
+        {
+          "spring.datasource.url": "jdbc:mysql://mysqldb:3306/${MYSQL_DATABASE}",
+          "spring.datasource.username": "${MYSQL_ROOT_USER}",
+          "spring.datasource.password": "${MYSQL_ROOT_PASSWORD}",
+          "spring.jpa.properties.hibernate.dialect": "org.hibernate.dialect.MySQLDialect"
+        }
+      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:8080"
```
Comment (liliia-ponomarenko) [inline docker-compose.yml:32]: you can use DEBUG_PORT variable

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! Minor comment

**Pull request merged**